string_grouper/string_grouper.py

```diff
@@ -212,36 +131,135 @@ def __init__(self, master: pd.Series,
         # After the StringGrouper is built, _matches_list will contain the indices and similarities of the matches
         self._matches_list: pd.DataFrame = pd.DataFrame()
         # _true_max_n_matches will contain the true maximum number of matches over all strings in master if
         # self._config.min_similarity <= 0
         self._true_max_n_matches = None
+        self._vectorizer = self._fit_vectorizer()
+        self.is_build = False  # indicates if the grouper was fit or not
+
+    def reset_data(self,
+                   master: pd.Series,
+                   duplicates: Optional[pd.Series] = None,
+                   master_id: Optional[pd.Series] = None,
+                   duplicates_id: Optional[pd.Series] = None):
+        """
+        Sets the input Series of a StringGrouper instance without changing the underlying corpus.
+        :param master: pandas.Series. A Series of strings in which similar strings are searched, either against itself
+        or against the `duplicates` Series.
+        :param duplicates: pandas.Series. If set, for each string in duplicates a similar string is searched in Master.
+        :param master_id: pandas.Series. If set, contains ID values for each row in master Series.
+        :param duplicates_id: pandas.Series. If set, contains ID values for each row in duplicates Series.
+        :param kwargs: All other keyword arguments are passed to StringGrouperConfig
+        """
+        self._set_data(master, duplicates, master_id, duplicates_id)
+
+    def clear_data(self):
+        self._master = None
+        self._duplicates = None
+        self._master_id = None
+        self._duplicates_id = None
+        self._matches_list = None
+        self._left_DataFrame = None
+        self._right_DataFrame = None
+        self.is_build = False
+
+    def update_options(self, **kwargs):
+        """
+        Updates the kwargs of a StringGrouper object
+        :param **kwargs: any StringGrouper keyword=value argument pairs
+        """
+        _ = StringGrouperConfig(**kwargs)
+        old_kwargs = self._config._asdict()
+        old_kwargs.update(kwargs)
+        self._set_options(**old_kwargs)
+
+    @property
+    def master(self):
+        return self._master
+
+    @master.setter
+    def master(self, master):
+        if not StringGrouper._is_series_of_strings(master):
+            raise TypeError('Master input does not consist of pandas.Series containing only Strings')
+        self._master = master
+
+    @property
+    def duplicates(self):
+        return self._duplicates
+
+    @duplicates.setter
+    def duplicates(self, duplicates):
+        if duplicates is not None and not StringGrouper._is_series_of_strings(duplicates):
+            raise TypeError('Duplicates input does not consist of pandas.Series containing only Strings')
+        self._duplicates = duplicates
+
+    @property
+    def master_id(self):
```

Comment:
Should we just make these attributes public? So instead of having `self._master_id` we get self.master_id. This eliminates the need for having these setter and property functions.

Reply:
@Bergvca 

Sure.  `master_id` and `duplicates_id` do not need setter and getter functions as neither of them is validated on its own.  I agree that we can get rid of them altogether.  Never mind making them public.  Actually it is best they are always input together with `master` and `duplicates` through the `__init__()` or `reset_data()` functions.